File: app/src/main/java/com/simon/harmonichackernews/utils/StoryUpdate.java
package com.simon.harmonichackernews.utils;

import com.simon.harmonichackernews.data.Story;

public class StoryUpdate {

    public static StoryUpdateListener storyUpdateListener;

    public static void updateStory(Story story) {
        if (storyUpdateListener != null) {
            storyUpdateListener.callback(story);
        }
    }

    public static void setStoryUpdatedListener(StoryUpdateListener storyUpdateListener) {
        StoryUpdate.storyUpdateListener = storyUpdateListener;
    }

    public static void clearStoryUpdatedListener(StoryUpdateListener storyUpdateListener) {
        if (StoryUpdate.storyUpdateListener == storyUpdateListener) {
            StoryUpdate.storyUpdateListener = null;
        }
    }

    public interface StoryUpdateListener {
        void callback(Story story);
    }

}

File: app/src/main/java/com/simon/harmonichackernews/utils/StoryUpdate.kt
package com.simon.harmonichackernews.utils

import com.simon.harmonichackernews.data.Story

object StoryUpdate {
    var storyUpdateListener: StoryUpdateListener? = null

    fun updateStory(story: Story?) {
        if (storyUpdateListener != null) {
            storyUpdateListener!!.callback(story)
        }
    }

    fun setStoryUpdatedListener(storyUpdateListener: StoryUpdateListener?) {
        StoryUpdate.storyUpdateListener = storyUpdateListener
    }

    fun clearStoryUpdatedListener(storyUpdateListener: StoryUpdateListener?) {
        if (StoryUpdate.storyUpdateListener === storyUpdateListener) {
            StoryUpdate.storyUpdateListener = null
        }
    }

    fun interface StoryUpdateListener {
        fun callback(story: Story?)
    }
}
